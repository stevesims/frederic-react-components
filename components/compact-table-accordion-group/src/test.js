import React from 'react';
import { mount, shallow } from 'enzyme';
import Component from '.';

describe('CompactTableAccordionGroup', () => {
  let wrapper;
  const changeHandler = jest.fn();

  it('renders without crashing', () => {
    wrapper = shallow(<Component expanded="Expanded">Test</Component>);
  });

  it('renders title when passed', () => {
    wrapper = shallow(
      <Component
        expanded="Expanded"
        onChange={changeHandler}
        title="Example title"
      >
        Test
      </Component>,
    );
    expect(
      wrapper
        .find('Title')
        .childAt(0)
        .text(),
    ).toBe('Example title');
  });

  it('triggers onChange when open button is clicked', () => {
    wrapper = mount(
      <Component expanded="Expanded" onChange={changeHandler}>
        Test
      </Component>,
    );
    wrapper.find('button').simulate('click');
    expect(changeHandler).toHaveBeenCalledTimes(1);
  });

  it('does not trigger onChange when title is clicked and changeOnTitleClick is false', () => {
    changeHandler.mockReset();
    wrapper = mount(
      <Component
        expanded="Expanded"
        title="Example title"
        onChange={changeHandler}
      >
        Test
      </Component>,
    );
    wrapper.find('header').simulate('click');
    expect(changeHandler).not.toHaveBeenCalled();
  });

  it('triggers onChange when title is clicked and changeOnTitleClick is true', () => {
    changeHandler.mockReset();
    wrapper = mount(
      <Component
        expanded="Expanded"
        title="Example title"
        onChange={changeHandler}
        changeOnTitleClick
      >
        Test
      </Component>,
    );
    wrapper.find('header').simulate('click');
    expect(changeHandler).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
