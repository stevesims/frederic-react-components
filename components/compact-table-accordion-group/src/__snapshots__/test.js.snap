// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompactTableAccordionGroup matches snapshot 1`] = `
.emotion-5 {
  font-family: Roboto,Arial,sans-serif;
  font-size: 14px;
  padding-bottom: 10px;
  padding-top: 10px;
  position: relative;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 8px;
}

.emotion-0 {
  background: none;
  border: none;
  cursor: pointer;
  height: 20px;
  padding: 0;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  width: 20px;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.emotion-0:focus {
  outline: none;
}

.emotion-1 {
  cursor: pointer;
  padding-left: 16px;
}

<CompactTableAccordionGroup
  changeOnTitleClick={true}
  expanded="Expanded"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "open": true,
          },
        ],
      ],
    }
  }
  open={false}
  title="Example title"
>
  <Container>
    <div
      className="emotion-5 emotion-6"
    >
      <TitleRow>
        <div
          className="emotion-3 emotion-4"
        >
          <OpenButton
            onChange={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "open": true,
                    },
                  ],
                ],
              }
            }
            open={false}
          >
            <Styled(button)
              onClick={[Function]}
              open={false}
            >
              <button
                className="emotion-0"
                onClick={[Function]}
                open={false}
              >
                <Arrow
                  fill="#0C60A2"
                  width={20}
                >
                  <svg
                    height="20px"
                    viewBox="0 0 6 4"
                    width="20px"
                  >
                    <polygon
                      fill="#0C60A2"
                      points="3 4 6 1 0 1"
                    />
                  </svg>
                </Arrow>
              </button>
            </Styled(button)>
          </OpenButton>
          <Title
            clickable={true}
            onClick={[Function]}
          >
            <header
              className="emotion-1 emotion-2"
              onClick={[Function]}
            >
              Example title
            </header>
          </Title>
        </div>
      </TitleRow>
      <Collapse
        fixedHeight={-1}
        forceInitialAnimation={false}
        hasNestedCollapse={false}
        isOpened={false}
        onMeasure={[Function]}
        onRender={[Function]}
        onRest={[Function]}
        style={Object {}}
        theme={
          Object {
            "collapse": "ReactCollapse--collapse",
            "content": "ReactCollapse--content",
          }
        }
      >
        <Motion
          defaultStyle={
            Object {
              "height": 0,
            }
          }
          onRest={[Function]}
          style={
            Object {
              "height": 0,
            }
          }
        >
          <div
            className="ReactCollapse--collapse"
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
              }
            }
          >
            <div
              className="ReactCollapse--content"
            >
              Expanded
            </div>
          </div>
        </Motion>
      </Collapse>
    </div>
  </Container>
</CompactTableAccordionGroup>
`;
